### PR TITLE
Settings functionality

### PR DESCRIPTION
**Pull request by avikalpg**
- RepoList displays the actual configuration of the repository according to the database
- Changing the switch calls an API to change the configuration and displays the updated state if the API call was successful

### CODE REVIEW

File: pages/api/setRepoConfig.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	repo: RepoIdentifier,
+	configType: 'auto_assign' | 'comment',
+	value: boolean
+}
+
+const isSetRepoConfigReqBody = (x: any): x is SetRepoConfigReqBody => {
+	return x && isRepoIdentifier(x.repo) && (x.configType === "auto_assign" || x.configType === "comment") && typeof x.value === "boolean";
+}
+
+export default function handler(
```
Comment (tapishr) [inline pages/api/setRepoConfig.ts:15]: Use const and a better name than handler

File: utils/db/repos.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import conn from '.';
+import type { RepoIdentifier } from '../../types/repository';
+import { convert } from './converter';
+
+export const setRepoConfig = async (repo: RepoIdentifier, configType: 'auto_assign' | 'comment', value: boolean) => {
```
Comment (tapishr) [inline utils/db/repos.ts:5]: everytime we add a new type of setting, will we have to change the type from 'auto_assign' | 'comment' ? We might have a lot of settings, which could be more complex (setting inside a sub-setting etc.)

Comment (avikalpg) [reply utils/db/repos.ts:5]: Whenever that happens (a new setting is added), a lot of work will need to be done. In my opinion, a statically defined type must be modified for that. 

And if we introduce more complex settings (sub settings etc.) we will have to modify the entire UI as well. 

If this type is used in many places, we can make a dedicated TS type for it. I am not sure what else can be done to improve this. But if you have any recommendations, please let me know.

File: pages/api/setRepoConfig.ts (hunk 1 above)
Comment (avikalpg) [reply pages/api/setRepoConfig.ts]: Fixed.

File: utils/db/repos.ts (hunk 2 above)
Comment (tapishr) [reply utils/db/repos.ts:5]: Cool, we'll burn that bridge when we have to cross it

**Review by tapishr — APPROVED** [review]

**Pull request merged**